driver/lcd_common: rename from ili9341 driver

fjmolinas opened the pull request:
### Contribution description

A year ago I wrote an st7735 driver that was based on the ili9341, most of the code is exactly the same though.

LCD controllers ST7735 and ili9341 share very similar functions, the main differences are related to the set of available commands and initialization procedure. So instead of adding a new driver for the st77335 I rename the ili9341 driver to a common lcd driver. Only available commands and initialization procedure are a bit different.

Other changes:
- add LCD_COMMON_OFFSET_X/Y: ST7735 display might have an offset depending on rotation and size
- allow defining the default rotation

There is a lot of renaming in this PR but `ili9341` code should be mostly exactly the same, there is a very small difference in binary size mainly because "puts" have changed and the extra `CONFIG_%`

```
"make" -C /home/francisco/workspace/riot-firmwares/RIOT/sys/xtimer
   text    data     bss     dec     hex filename
  30332     112    2340   32784    8010 /home/francisco/workspace/riot-firmwares/RIOT/tests/driver_lcd_common/bin/stm32f429i-disc1/tests_driver_lcd_common.elf
```
```
   text    data     bss     dec     hex filename
  30300     112    2340   32752    7ff0 /home/francisco/workspace/riot-firmwares/RIOT/tests/driver_ili9341/bin/stm32f429i-disc1/tests_driver_ili9341.elf
```

### Testing procedure

- `make -C tests/driver_lcd_common flash`
- `LCD_CONTROLLER=st7735 make -C tests/driver_lcd_common flash`
- `make -C tests/disp_dev flash`
- `make -C tests/pkg_lvgl flash`

aabadie commented on drivers/lcd_common/Makefile.dep:
```diff
@@ -0,0 +1,11 @@
+FEATURES_REQUIRED += periph_gpio
+USEMODULE += xtimer
+
+DEFAULT_MODULE += ili9341
+ifneq (,$(filter ili9341,$(USEMODULE)))
+  DISABLE_MODULE += st7735
+endif
+ifneq (,$(filter st7735,$(USEMODULE)))
+  DISABLE_MODULE += ili9341
+endif
```
I don't really understand the need for a DEFAULT_MODULE here. A simple direct dependency seems enough: if the user wants to use ili9341, then lcd_common is pulled in automatically.

Maybe it's because the current approach prevents one ili9341 and one st7735 display module to be used at the same time ?

fjmolinas commented:
@aabadie I re-worked this PR quite a bit, now it's compatible with both drivers, I tested this by running `tests/disp_dev` on a board with a `st7735` and `ili9341` based display. Some things were also changed:

- rgb_channels: added as a new parameter. Why is this not compile time? because you can have multiple screens with different sizes and that needs to be handled.
- st7735: there is a x/y_offset  that is not compile time for the same reason.

There are maybe still things to be improved regarding code duplicity, but please tell me what you think of the current approach.

aabadie commented:
> please tell me what you think of the current approach

I don't  remember how it was before but looking at the current approach, I must say that I don't fully understand all changes related to the lcd common. It seems to me that it introduces another layer of abstraction and thus is kind of redundant with the disp_dev purpose. It also adds another layer of auto_init code (with driver specific code inside) that I don't like.
Why not just keep the lcd common as an external module where function are directly used by the specific ili9341/st7785 drivers ? And use disp_dev as the first attribute of each driver specific code, instead of lcd.

But anyway, with all recent changes with disp_dev test application and build system enhancement, you have to rebase and rework this again.

fjmolinas commented:
> But anyway, with all recent changes with disp_dev test application and build system enhancement, you have to rebase and rework this again.

I rebased it now.

fjmolinas commented on tests/saul/modules.kconfig.txt:
```diff
@@ -0,0 +1,40 @@
+=== [ATTENTION] Testing Kconfig dependency modelling ===
+=== [ATTENTION] Testing Kconfig dependency modelling ===
+make: Nothing to be done for 'clean'.
```
should be removed

fjmolinas commented:
@aabadie I remember you not liking this approach initially, I would rather not merge something where the architecture is not liked just because it works.

aabadie commented:
> I remember you not liking this approach initially,

I think it was https://github.com/RIOT-OS/RIOT/pull/16176#issuecomment-851927807

> I would rather not merge something where the architecture is not liked just because it works.

No, I like it because it works :)
More seriously, I think the generic lcd driver is nice but I don't really like the fact that the logic was pushed so much that there's an `auto_init_lcd` that wraps the `auto_init_{ili9341,st735}` and also a common params structure, even if they are very similar. This somehow breaks the device driver implementation model (for the auto_init_part) and the generic lcd implementation cannot be instanciated alone, you need an implementation of a concrete device (ili9341/st7735), this can be confusing.

For the moment, I can't clearly think of a better design, maybe keeping this one (generic lcd attribute in each device descriptor) but not the `auto_init_lcd` and `lcd_params` would be ok since it would kind of prevent the lcd api to be used directly.

aabadie commented:
@fjmolinas I've updated a bit your PR to something that I think is better. Tested on adafruit-clue (ili9341) and adafruit-pybadge (with #17903 included) and both are working as expected.
Main changes I added:
- split the `auto_init_lcd` code in `auto_init_9341` and `auto_init_st7735`
- remove the use of symlink between `tests/driver_st7735` and `driver_ili9341`
- the lcd driver struct is now called `lcd_driver_t` instead of `lcd_desc_t`

If you are ok with my changes, then I think this is ready to go.

aabadie reviewed (approved):
Code changes are good enough to me now. @fjmolinas is ok to merge this one as it is now.

ACK
